frame/assets/src/functions.rs

```diff
@@ -470,8 +471,8 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 				account.balance = account.balance.saturating_sub(actual);
 				if account.balance < details.min_balance {
 					debug_assert!(account.balance.is_zero(), "checked in prep; qed");
-					if let Remove =
-						Self::dead_account(id, target, &mut details, &account.reason, false)
+					target_died = Remove;
+					if let Remove = Self::dead_account(target, &mut details, &account.reason, false)
```

Comment:
here in case the account is actually not dead because it hold a deposit then we don't want to execute Freezer::died.

Before the PR it was the case, this PR is changing it AFAICT.

Reply:
The `dead_account` function currently always runs `died`, except when a balance is held **and** `force == false`. So this code is still wrong, but maybe for a different reason than you thought (will fix).

Just so we don't misunderstand each other here;  
the `return Ok(())` in the `if` removes the account, since it was set to `None` by the `maybe_account.take()` call above (might worth refactoring).

Reply:
here `force` is false. that is why I didn't say it.

My point is that for this function:
* before this PR an account which goes below min balance but is kept because its existential reason is `DepositHeld` doesn't execute `FrozenBalance::died`, it will execute it when the account is refund.
* with this PR an account which goes below min balance but is kept because its existential reason is `DepositHeld` does execute `FrozenBalance::died`, it will also execute it when the account is refund.

Reply:
I think the simplest solution is to put the result of dead_account call into the variable as written in https://github.com/paritytech/substrate/pull/10473/files#r768322048

Reply:
Done